server/src/visitors/YmlBaseVisitor.ts

```diff
@@ -1,20 +1,242 @@
+        left: ParserRuleContext,
+        centralNodeStart: number,
+        centralNodeEnd: number,
+        right: ParserRuleContext,
+        leftAndStartSameLine: boolean,
+        endAndRightSameLine: boolean,
+    ) {
+        const leftEnd = left.stop;
+        const rightStart = right.start;
+
+        this.setOneSpaceInterval(leftEnd.stopIndex, centralNodeStart, leftAndStartSameLine);
+        this.setOneSpaceInterval(centralNodeEnd, rightStart.startIndex, endAndRightSameLine);
+    }
+
+    /**
+     * Add text edits in order to have no space between `leftElementEnd` and `rightElementStart`.
+     * Do nothing if there is already a zero space distance.
+     */
+    private removeSpaceInterval(leftElementEnd: number, rightElementStart: number) {
+        if (rightElementStart - leftElementEnd === 1) {
+            // There is already no space between the two elements.
+            return;
+        }
+        // One or more spaces.
+        this.filePossibleEdits.push(
+            TextEdit.del(
+                Range.create(this.document.positionAt(leftElementEnd + 1), this.document.positionAt(rightElementStart)),
+            ),
+        );
+    }
+
+    /**
+     * Add text edits in order to have only one space between `leftElementEnd` and `rightElementStart`.
+     * Do nothing if there is already a one space distance or if the two elements are not on the same line.
+     */
+    private setOneSpaceInterval(leftElementEnd: number, rightElementStart: number, sameLine: boolean): void {
+        // Don't force one space interval when the left element and the right element are not on the same line.
+        if (!sameLine) {
+            return;
+        }
+        if (rightElementStart - leftElementEnd === 2) {
+            // There is already one space.
+            return;
+        }
+        if (rightElementStart - leftElementEnd === 1) {
+            // Add missing space.
+            // Case where we have something like “a== b” or “a==b”,
+            // the distance between the end of “a“ (leftElementEnd) and the first “=” (rightElementStart)
+            // is 1 because there is no other space between them.
+            this.filePossibleEdits.push(TextEdit.insert(this.document.positionAt(leftElementEnd + 1), ' '));
+            return;
+        }
+        // More than one space, we replace all the spaces with only one.
+        this.filePossibleEdits.push(
+            TextEdit.replace(
+                Range.create(this.document.positionAt(leftElementEnd + 1), this.document.positionAt(rightElementStart)),
+                ' ',
+            ),
+        );
+    }
+
+    /** Add an edit that adds a semicolon right after the provided node. */
+    private addSemiColon(node: ParserRuleContext): void {
+        this.filePossibleEdits.push(TextEdit.insert(this.document.positionAt(node.stop.stopIndex + 1), ';'));
+    }
+
+    /** Indicate if the document format is impossible and shouldn't be done. */
```

Comment:
The docs and function name didn’t make me expect to find `docFormatSettings.enableDocumentFormat` here.

Reply:
This doesn't bother me.

Reply:
I mean, “impossible” for something that the user delibirately disabled? Almost makes it sound like the user is struggling against the piece of software rather than using it.

Reply:
It's OK, we don't use `docFormatSettings.enableDocumentFormat` here anymore.